python/lsst/ip/isr/isrTask.py

```diff
@@ -585,6 +594,11 @@ def run(self, ccdExposure, bias=None, linearizer=None, dark=None, flat=None, def
                 self.maskAndInterpNan(ccdExposure)
                 interpolationDone = True
 
+            if self.config.brighterFatterLevel == 'DETECTOR':
+                bfKernel = bfKernel[ccdExposure.getDetector().getId()]
```

Comment:
You are replacing `bfKernel` with an element from the variable of the same name. I think this would be a lot less confusing if you used a different variable name for the sequence than for the element of the sequence, e.g. `kernelElement = brighterFatterKernel[...]`

It would also be less confusing if the dataset type name was not singular (e.g. `brighterFatterKernels`) since you are getting a kernel from it. But I suspect it's too late to reasonably do anything about that.

Reply:
Also, as a minor point: since "bf" is expanded almost everywhere else, the code would be a bit easier to read if you used "brighterFatter" for the kernel variable as well -- especially as it matches a dataset type.

Of course it gets long if you need two names for the kernel vs. the collection of kernels...perhaps just leave out "bf" for the former, e.g. `detectorKernel = brighterFatterKernel[...]`

Reply:
Oh good lord, that's a great spot there, overwriting the object with an element of it, that was totally unintended! I have changed that now.

Regarding the rest, I take your point, but sadly the name `bfKernel` is passed in as a kwarg so changing that is RFC territory (in reality I'm sure it's fine, but _technically_ I think it's required?). So as you say, I don't think it's worth changing now, especially as all of isr will need a cleanup at some point.